### PR TITLE
FOUR-8559 Refactor throwMessageEvent function to use Nayra BPMN engine

### PR DESCRIPTION
**Pull request by caleeli**
This PR refactors the throwMessageEvent function. The primary objective was to enhance the function's performance and provide it the capability to trigger a BpmnAction message event using instance ID, element ID, message reference, and an optional payload. This is important to improve the Webhook Callback in our package-data-sources.

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-8559

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-6e88847f56.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-6e88847f56.eng.processmaker.net

File: ProcessMaker/Nayra/Managers/WorkflowManagerRabbitMq.php [hunk 1]
```diff
@@ -138,7 +141,7 @@ public function completeCatchEvent(Definitions $definitions, ExecutionInstanceIn
         $this->validateData($data, $definitions, $element);
 
         // Get complementary information
-        $version = $definitions->getLatestVersion();
+        $version = $instance->process_version_id;
```
Comment (julceslauhub) [inline ProcessMaker/Nayra/Managers/WorkflowManagerRabbitMq.php:144]: you need to change the line: `'bpmn' => $version->getKey(),` to `'bpmn' => $version,`

Comment (caleeli) [reply ProcessMaker/Nayra/Managers/WorkflowManagerRabbitMq.php:144]: Nice catch, fixed.

**Review by julceslauhub — APPROVED** [review]
Approved

**Pull request merged**